src/libraries/System.IO.FileSystem/tests/Directory/GetFileSystemEntries_str.cs

```diff
@@ -205,26 +204,31 @@ public void HiddenFilesAreReturned()
         #region PlatformSpecific
 
         [Fact]
-        public void InvalidPath_Core()
+        public void InvalidPath()
         {
-            foreach (char invalid in Path.GetInvalidFileNameChars())
+            Assert.All(Path.GetInvalidFileNameChars(), invalidChar =>
             {
-                string badPath = string.Format($"{TestDirectory}{Path.DirectorySeparatorChar}te{invalid}st");
-                switch (invalid)
+                string badPath = string.Format($"{TestDirectory}{Path.DirectorySeparatorChar}te{invalidChar}st");
+
+                switch (invalidChar)
                 {
                     case '/':
                     case '\\':
-                    case ':':
                         Assert.Throws<DirectoryNotFoundException>(() => GetEntries(badPath));
                         break;
+                    case ':':
+                        // Framework does not support reading colon
+                        AssertExtensions.Throws<DirectoryNotFoundException, NotSupportedException>(() => GetEntries(badPath));
+                        break;
                     case '\0':
                         Assert.Throws<ArgumentException>(() => GetEntries(badPath));
                         break;
                     default:
-                        Assert.Throws<IOException>(() => GetEntries(badPath));
+                        // Same exception message, different exception type
```

Comment:
These tests that say "Same exception message, different exception type" aren't actually validating a message.

Reply:
Can you show me how to validate the exception message string, if it's possible? I vaguely recall from a past PR that attempting to compare exception messages is not possible in unit tests because of localization. I can try to find the comment.

Reply:
> Can you show me how to validate the exception message string, if it's possible? I vaguely recall from a past PR that attempting 

It's possible, but we generally don't do it. My comment was more about the fact that the comment is suggesting we're validating something we're not. We can delete the comment.